src/jwks.cc

```diff
@@ -85,43 +80,41 @@ class KeyGetter : public WithStatus {
       updateStatus(Status::JwksEcCreateKeyFail);
       return nullptr;
     }
-    BIGNUM *bn_x = createBigNumFromBase64UrlString(x);
-    BIGNUM *bn_y = createBigNumFromBase64UrlString(y);
+    bssl::UniquePtr<BIGNUM> bn_x = createBigNumFromBase64UrlString(x);
+    bssl::UniquePtr<BIGNUM> bn_y = createBigNumFromBase64UrlString(y);
     if (!bn_x || !bn_y) {
       // EC public key field x or y Base64 decode fail
       updateStatus(Status::JwksEcXorYBadBase64);
       return nullptr;
     }
 
-    if (EC_KEY_set_public_key_affine_coordinates(ec_key.get(), bn_x,
-                                                 bn_y) == 0) {
+    if (EC_KEY_set_public_key_affine_coordinates(ec_key.get(), bn_x.get(),
+                                                 bn_y.get()) == 0) {
       updateStatus(Status::JwksEcParseError);
       return nullptr;
     }
     return ec_key;
   }
 
   bssl::UniquePtr<RSA> createRsaFromJwk(const std::string& n,
-                                          const std::string& e) {
+                                        const std::string& e) {
     bssl::UniquePtr<RSA> rsa(RSA_new());
-    BIGNUM *bn_n = createBigNumFromBase64UrlString(n);
-    BIGNUM *bn_e = createBigNumFromBase64UrlString(e);
+    bssl::UniquePtr<BIGNUM> bn_n = createBigNumFromBase64UrlString(n);
+    bssl::UniquePtr<BIGNUM> bn_e = createBigNumFromBase64UrlString(e);
 
     if (bn_n == nullptr || bn_e == nullptr) {
```

Comment:
do you need to check using `.get()` here?

Reply:
No, there is an overloaded `==` operator that specifically takes `unique_ptr` & `nullptr` arguments, for use like this.